### PR TITLE
Allow running tests against API without cassette recording

### PR DESCRIPTION
**Pull request by bkabrda**
### What does this PR do?

This PR allows setting `RECORD=none` which runs the tests against the API without touching cassettes at all.

### Review checklist
Please check relevant items below:
- [ ] This PR includes all [newly recorded cassettes](/DEVELOPMENT.md) for any modified tests.

- [ ] This PR does not rely on API client schema changes.
    - [ ] The CI should be fully passing.
- [ ] Or, this PR relies on API schema changes and this is a Draft PR that includes tests.
  - Note: CI shouldn't be run on this Draft PR, as its expected to fail without the corresponding schema changes.

### CODE REVIEW

File: .azure-pipelines/all.yml [hunk 1]
```diff
@@ -59,7 +59,7 @@ jobs:
         DD_TEST_CLIENT_APP_KEY: $(datadogAPPKey)
         DD_TRACE_ANALYTICS_ENABLED: "true"
         DD_TRACE_ANNOTATIONS: org.junit.Test
-        RECORD: "true"
+        RECORD: "none"
```
Comment (jirikuncar) [inline .azure-pipelines/all.yml:62]: 👍

**Comment by bkabrda** [comment]
Thanks for the review, merging.

**Pull request merged**